clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -61,7 +61,7 @@ public static void registerAppInfo(String prefix, String id) {
         }
     }
 
-    public static void unregisterAppInfo(String prefix, String id) {
+    public static synchronized void unregisterAppInfo(String prefix, String id) {
```

Comment:
Btw, we should take the chance and make `version` and `commitId` final. Something like:

```java
private static final String version;
    private static final String commitId;

    static {
        Properties props = new Properties();
        try (InputStream resourceStream = AppInfoParser.class.getResourceAsStream("/kafka/kafka-version.properties")) {
            props.load(resourceStream);
        } catch (Exception e) {
            log.warn("Error while loading kafka-version.properties :" + e.getMessage());
        }
        version = props.getProperty("version", "unknown").trim();
        commitId = props.getProperty("commitId", "unknown").trim();
    }
```